### PR TITLE
sphinxcontrib-moderncmakedomain init at 3.29.0

### PR DESCRIPTION
**Pull request by jhol**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+, fetchPypi
+, isPy27
+, hatchling
+, sphinx
+}:
+
+buildPythonPackage rec {
+  pname = "sphinxcontrib-moderncmakedomain";
+  version = "3.27.0";
+  disabled = isPy27;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix:12]:
```suggestion
```
python2 is so EOL, we don't need to care

File: pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    sphinx
+  ];
+
+  pythonNamespaces = [ "sphinxcontrib" ];
+
+  meta = with lib; {
+    description = "sphinxcontrib-moderncmakedomain is a Sphinx extension which renders CMake documentation.";
+    homepage = "https://github.com/scikit-build/moderncmakedomain";
+    license = licenses.bsd3;
+    maintainers = teams.sphinx.members;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix:32]: This probably belongs more to the scikit people than the general sphinx

File: pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  nativeBuildInputs = [
+    hatchling
+    sphinx
+  ];
+
+  pythonNamespaces = [ "sphinxcontrib" ];
+
+  meta = with lib; {
+    description = "sphinxcontrib-moderncmakedomain is a Sphinx extension which renders CMake documentation.";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix:29]:
```suggestion
    description = "Sphinx extension which renders CMake documentation";
```

File: pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    inherit version;
+    pname = "sphinxcontrib_moderncmakedomain";
+    hash = "sha256-UeJZ6R9Y0XzA+skwf9QBBqpZ1ayqdBiHkD/DZgNh0aE=";
+  };
+
+  nativeBuildInputs = [
+    hatchling
+    sphinx
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix:25]: Please run the tests https://github.com/scikit-build/moderncmakedomain/tree/main/tests

File: pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix (hunk 3 above)
Comment (jhol) [reply pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix]: Done!

File: pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix (hunk 4 above)
Comment (jhol) [reply pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix:25]: Done!

File: pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix:44]: then I am not sure but sphinx members should not receive this.

Comment (jhol) [reply pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix]: I set myself as the maintainer! I'll keep it updated.

File: pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    pytest
+    sphinx
+  ];
+
+  checkPhase = ''
+    runHook preCheck
+    pytest tests/
+    runHook postCheck
+  '';
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix:37]:
```suggestion
    pytestCheckHook
    sphinx
  ];

```

**Comment by jhol** [comment]
I think this should be good to go now. Is there anything I can do to help get this accepted?

File: pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  src = fetchPypi {
+    inherit version;
+    pname = "sphinxcontrib_moderncmakedomain";
+    hash = "sha256-NYfe8kH/JXfQu+8RgQoILp3sG3ij1LSgZiQLXz3BtbI=";
+  };
+
+  nativeBuildInputs = [
+    hatchling
+    sphinx
+  ];
```
Comment (natsukium) [inline pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix:25]:
```suggestion
  build-system = [ hatchling ];
  
  dependencies = [ sphinx ];
```
We use `build-system` and `dependencies` for python dependencies.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

Also, following the upstream, `sphinx` should go to `dependencies`.
https://github.com/scikit-build/moderncmakedomain/blob/b4cdb0089ad40189f4cffeab7f859265da599336/pyproject.toml#L40-L42

File: pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+  nativeCheckInputs = [
+    defusedxml
+    pytestCheckHook
+    sphinx
```
Comment (natsukium) [inline pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix:30]:
```suggestion
```

File: pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  ];
+
+  pythonNamespaces = [ "sphinxcontrib" ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix:34]: Could you enable `pythonImportsCheck`?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck-using-pythonimportscheck

Comment (jhol) [reply pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix:32]: Sounds good. Thanks for the feedback

**Comment by jhol** [comment]
I think this is good to do. Is there anything else that needs to be done to move this forward?

**Comment by jhol** [comment]
Just rebased to fix merge conflict in maintainers file.

File: pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix (hunk 6 above)
[the remark this answers is not in the document]
Comment (jhol) [reply pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix]: Done! Thanks for the feedback.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by jhol** [comment]
:tada:

**Pull request merged**